lib/MusicBrainz/Server/Data/Recording.pm

```diff
@@ -422,9 +422,13 @@ sub find_recent_by_artists
 {
     my ($self, $artist_ids) = @_;
 
-    my $search_url = sprintf("http://%s/ws/2/recording/?query=title:\"\"",
-                              DBDefs->LUCENE_SERVER);
-
+    if (DBDefs->SEARCH_ENGINE eq 'LUCENE') {
+        my $search_url = sprintf("http://%s/ws/2/recording/?query=title:\"\"",
```

Comment:
Shouldn't `$search_url` declared before the `if`/`else` since it is used outside the scope of each branch ?

```perl
$my search_url;
if (...) {
   $search_url = 'A....';
} else {
   $search_url = 'B....';
}

do_something($search_url);
```

Reply:
Yes, unlike python perl has lexical scope, so `$search_url` needs to be declared outside the `if`.

Reply:
Ah, let me fix that.